react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+class ValidatedInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      password: '',
+      error: '',
+      icon: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ password: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    if (this.state.password === '') {
+      this.setState({ icon: 'fas fa-times error-icon ms-2' });
+      this.setState({ error: 'A password is required.' });
+    } else if (this.state.password.length < 8) {
+      this.setState({ icon: 'fas fa-times error-icon ms-2' });
+      this.setState({ error: 'Your password is too short.' });
+    } else {
+      this.setState({ icon: 'fas fa-check success-icon ms-2' });
```

Comment:
since the error icon only appears if error is not an empty string, then you can leave out the `icon` state entirely and determine the `icon` in `render()`.